### PR TITLE
feat: verify mocks in the CI

### PR DESCRIPTION
**Pull request by aymericDD**
## What does this PR do?

- [x] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- `Verify if mocks are updated.`

## Code Quality Checklist

- [x] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -273,6 +273,11 @@ workflows:
           target: manifests
           diff-fails: true
           requires: [prepare-env]
+      - go-make:
+          name: mocks
+          target: generate-mocks
```
Comment (luphaz) [inline .circleci/config.yml:278]: I would not expect mock to self generate/update on their own in the CI

Usually, you should instead FAIL if mocks are not properly generated (compare with git diff and exit)

Comment (aymericDD) [reply .circleci/config.yml:278]: How can I ensure the mocks are updatdate if I don't generate them in the CI and compare the diff' ?

Comment (luphaz) [reply .circleci/config.yml:279]: The PR as is was not performing any validation or diff, correct?

Let me rephrase then

> I would not expect mock to self generate/update on their own in the CI


I would not expect mock to ONLY self generate/update on their own in the CI WITHOUT performing any follow up assertion.

Comment (luphaz) [reply .circleci/config.yml:279]: Indeed I misinterpreted the configuration file 😄 

Thanks for clarifying

File: .circleci/config.yml [hunk 2]
```diff
@@ -273,6 +273,12 @@ workflows:
           target: manifests
           diff-fails: true
           requires: [prepare-env]
+      - go-make:
+          executor: ubuntu
```
Comment (ptnapoleon) [inline .circleci/config.yml:277]: The other checks above run on the executor `golang`, no? I think they inherit from prepare-env. Does it still fail if you don't specify an executor here?

Comment (ptnapoleon) [inline .circleci/config.yml:277]:
```suggestion
```
testing ci without a specified executor...

Comment (ptnapoleon) [reply .circleci/config.yml]: oof, the `go` deps are present now, as expected, but because we run `make header-fix` after generating mocks, we also need our python deps. Not sure how to handle that, as they're on different executors. I see why you were planning to go with `ubuntu` here

**Review by ptnapoleon — APPROVED** [review]
